src/2048-expert/index.js

```diff
@@ -213,6 +213,7 @@ function keyUpLeft() {
     moveLeft()
     generateNewTile()
     generateNewTile()
+    addColours()
```

Comment:
Instead of looping through all the tiles on the board after every move, you could optimise this by only updating the colours for the cells that are merged, when they are merged. You could implement this by calling `addColours(index)` inside `mergeHorizontal()` and `mergeVertical()`.

Reply:
This does not work when new tiles are generated. This can be optimized in further development